### PR TITLE
[driver] Allow -msve-vector-bits=<n>+ syntax to mean no maximum vscale (release_14x)

### PR DESCRIPTION
**Pull request by LittleMeepo**
Make Flang's behaviour consistent with Clang's.

### CODE REVIEW

**Comment by LittleMeepo** [comment]
The purpose of this patch is to behave in the same way as Clang.

**Comment by LittleMeepo** [comment]
Corresponding PR in flang
https://github.com/flang-compiler/flang/pull/1319

**Review by pawosm-arm — CHANGES_REQUESTED** [review]
It's good to reflect clang's behavior, but this commit is missing test cases supporting it...

**Comment by LittleMeepo** [comment]
> It's good to reflect clang's behavior, but this commit is missing test cases supporting it...

Done.

File: clang/lib/Driver/ToolChains/ClassicFlang.cpp [hunk 1]
```diff
@@ -1054,37 +1054,50 @@ void ClassicFlang::ConstructJob(Compilation &C, const JobAction &JA,
...
+    if (Val.equals("128") || Val.equals("256") || Val.equals("512") ||
+        Val.equals("1024") || Val.equals("2048") || Val.equals("128+") ||
+        Val.equals("256+") || Val.equals("512+") || Val.equals("1024+") ||
+        Val.equals("2048+")) {
+      unsigned Bits = 0;
+      if (Val.endswith("+"))
+        Val = Val.substr(0, Val.size() - 1);
+      else {
+        bool Invalid = Val.getAsInteger(10, Bits); (void)Invalid;
+        assert(!Invalid && "Failed to parse value");
```
Comment (bryanpkc) [inline clang/lib/Driver/ToolChains/ClassicFlang.cpp:1071]: Why not put the function call inside the `assert`? e.g.
```c++
assert(!Val.getAsInteger(10, Bits) && "invalid argument for -msve_vector_bits");
```

Comment (LittleMeepo) [reply clang/lib/Driver/ToolChains/ClassicFlang.cpp:1071]: That's OK. But I just keep the format in Clang. Line 1060~1081 is copied from Clang.cpp

Comment (bryanpkc) [reply clang/lib/Driver/ToolChains/ClassicFlang.cpp:1071]: OK.

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by bryanpkc** [comment]
@pawosm-arm Could you take another look?

**Review by pawosm-arm — APPROVED** [review]

**Comment by bryanpkc** [comment]
@shivaramaarao This is an AArch64-specific patch. Would you still like to take a look?

**Comment by LittleMeepo** [comment]
I realized Flang use `-vscale_range_min` and `-vscale_range_max` as parameters to parse argument, so I changed all the `-vscale-range-min` and `-vscale-range-max` to `-vscale_range_min` and `-vscale_range_max`.

**Comment by bryanpkc** [comment]
@LittleMeepo Can you submit a PR with the same change for the `release_15x` branch?

**Pull request merged**